mne/viz/evoked.py

```diff
@@ -931,102 +944,31 @@ def _plot_evoked_white(evoked, noise_cov, scalings=None, rank=None, show=True):
-        if 'meg' in rank:
-            raise ValueError('When not using SSS separate rank values for mag '
-                             'or grad must be passed separately.')
     evoked = evoked.copy()  # handle ref meg
     passive_idx = [idx for idx, proj in enumerate(evoked.info['projs'])
                    if not proj['active']]
     # either applied already or not-- else issue
     for idx in passive_idx[::-1]:  # reverse order so idx does not change
         evoked.del_proj(idx)
 
-    picks = pick_types(evoked.info, meg=True, eeg=True, ref_meg=False,
-                       exclude='bads')
-    evoked.pick_channels([evoked.ch_names[k] for k in picks])
-    # important to re-pick. will otherwise crash on systems with ref channels
-    # as first sensor block
-    picks = pick_types(evoked.info, meg=True, eeg=True, ref_meg=False,
-                       exclude='bads')
-
-    picks_list = _picks_by_type(evoked.info, meg_combined=has_sss)
-    if has_meg and has_sss:
-        # reduce ch_used to combined mag grad
-        ch_used = list(zip(*picks_list))[0]
-    # order pick list by ch_used (required for compat with plot_evoked)
-    picks_list = [x for x, y in sorted(zip(picks_list, ch_used))]
-    n_ch_used = len(ch_used)
-
-    # make sure we use the same rank estimates for GFP and whitening
-
-    picks_list2 = [k for k in picks_list]
-    # add meg picks if needed.
-    if 'grad' in evoked and 'mag' in evoked:
-        # append ("meg", picks_meg)
-        picks_list2 += _picks_by_type(evoked.info, meg_combined=True)
-
-    rank_list = []  # rank dict for each cov
-    for cov in noise_cov:
-        this_rank = {}
-        C = cov['data'].copy()
-        # assemble rank dict for this cov, such that we have meg
-        for ch_type, this_picks in picks_list2:
-            # if we have already estimates / values for mag/grad but not
-            # a value for meg, combine grad and mag.
-            if ('mag' in this_rank and 'grad' in this_rank and
-                    'meg' not in rank):
-                this_rank['meg'] = this_rank['mag'] + this_rank['grad']
-                # and we're done here
-                break
-
-            if rank.get(ch_type) is None:
-                this_info = pick_info(evoked.info, this_picks)
-                idx = np.ix_(this_picks, this_picks)
-                this_estimated_rank = _estimate_rank_meeg_cov(
-                    C[idx], this_info, scalings)
-                _check_estimated_rank(
-                    this_estimated_rank, this_picks, this_info, evoked,
-                    cov, ch_type, has_meg, has_sss)
-                this_rank[ch_type] = this_estimated_rank
-            elif rank.get(ch_type) is not None:
-                this_rank[ch_type] = rank[ch_type]
-
-        rank_list.append(this_rank)
+    evoked.pick_types(meg=True, eeg=True, ref_meg=False, exclude='bads')
+    n_ch_used, rank_list, picks_list, has_sss = _triage_rank_sss(
+        evoked.info, noise_cov, rank, scalings)
```

Comment:
@larsoner is this call replacing all the work of the helper functions and other lines you removed?

Reply:
Yes it was copy-paste to refactor + bug fixes